From 6de811e067063b76e32955af6834770cf34b27b4 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Tue, 7 Jan 2020 17:04:57 +0200
Subject: [PATCH 054/269] s32g: Extend virtual address space to 36-bit

We must be able to do a full 1:1 MMU mapping to (eventually) access
the high 2GB of physical memory.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4474
---
 plat/s32g/include/platform_def.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 846a837d1..b7e108e6a 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -36,7 +36,8 @@
 #define S32G_CACHE_WRITEBACK_SHIFT	6
 #define CACHE_WRITEBACK_GRANULE		(1 << S32G_CACHE_WRITEBACK_SHIFT)
 #define PLAT_PHY_ADDR_SPACE_SIZE        (1ull << 36)
-#define PLAT_VIRT_ADDR_SPACE_SIZE	(1ull << 32)
+/* We'll be doing a 1:1 mapping anyway */
+#define PLAT_VIRT_ADDR_SPACE_SIZE	(1ull << 36)
 
 #define PLATFORM_CORE_COUNT		4
 #define PLATFORM_CLUSTER_COUNT		2
@@ -86,7 +87,7 @@
 #define S32G_DDR_SIZE		(S32G_DDR0_END - S32G_DDR0_BASE)
 
 /* Protected zone at the very top of DDR for our future use */
-#define S32G_PMEM_END		(S32G_DDR0_BASE + S32G_DDR_SIZE)
+#define S32G_PMEM_END		(S32G_DDR0_END)
 #define S32G_PMEM_LEN		0x00200000	/* 2MB */
 #define S32G_PMEM_START		(S32G_PMEM_END - S32G_PMEM_LEN)
 
-- 
2.17.1

